### ISSUE TITLE
/etc overlay established too late?

### ISSUE THREAD

**Issue by tommy4st** [gamer-os/gamer-os#60]
Some changes to /etc are not effective. Probably because the overlay is mounted to late in the process.

For example I cannot add a secondary hard drive to fstab. I can actually add it, but it is not mounted because fstab is read before the overlay is up. The problem is not about the directory, because I just created on in the home directory and I can mount it later on manually, but not on startup.

Another example is systemd services: I cannot enable a service. Again I can actually enable it in the way that the link is created, but it is not processed.

GamerOS 10, default

**Comment by alkazar** [comment]
Thanks for filing.

This is known behaviour. In order to mount the /etc overlay early enough we need to make changes to the initramfs file.

I haven't prioritized this because the use cases I am targeting doesn't include making changes via the command line. I am happy to accept code contributions for this but it is not a priority for me at the moment.

**Comment by tommy4st** [comment]
I did some research. An easy but incomplete way to at least honor the fstab would be to call `systemctl daemon-reload`. This reruns all generators, which means it reads fstab again and setup the units for all mounts.

Probably better is of course to integrate the mount into mkinitcpio, which I tested a bit with and it seams to work in my vanilla Arch install. Now the question for me is: How to integrate the new files? I need to add 2 files to the build process or image: [/etc/initcpio/{hooks,install}/frzr-etc](https://github.com/gamer-os/gamer-os/files/4037356/initcpio.zip)
 (both are used by mkinitcpio to mount early) and mkinitcpio.conf needs to be modified: "overlay" needs to be added to MODULES and "frzr-etc" added to HOOKS.

Should I include them in the build.sh script? And is there a reason mkinitcpio.conf is not included, but is a separate file?
